Server/Server.Test/ServerConnectionTests.cpp

```diff
@@ -0,0 +1,362 @@
+#include <catch2/catch.hpp>
+
+#include <iostream>
+#include <future>
+#include <asio.hpp>
+#include <Server.hpp>
+
+#include <ArgumentParser.hpp>
+#include <Cryptography.hpp>
+
+#include <Network/Connection.hpp>
+#include <Network/Primitives.hpp>
+
+using asio::ip::tcp;
+using Network::Connection;
+using Network::Message;
+using Network::SafeQueue;
+using RTC = std::chrono::system_clock;
+
+const Message& messageAcceptor(Message& message, const Message::MessageType& messageType)
+{
+    message.mHeader.mMessageType = messageType;
+    message.mHeader.mTimestamp   = RTC::to_time_t(RTC::now());
+
+    return message;
+}
+
+TEST_CASE("Comparing port test")
+{
+    uint16_t port;
+    uint16_t checkoutPort = 65003;
+
+    SECTION("Check comaring port with limiting port")
+    {
+        REQUIRE((port <= checkoutPort) == true);
+        REQUIRE((port > checkoutPort) == false);
+    }
+}
+
+TEST_CASE("Server Test")
+{
+    uint16_t port;
+
+    REQUIRE(port <= 65003);
+    Server::Server server(port);
+
+    SECTION("Check stopping & starting the server")
+    {
+        REQUIRE_NOTHROW(server.start());
+        REQUIRE_NOTHROW(server.stop());
+    }
+}
+
+TEST_CASE("Fail Server Test")
+{
+    uint32_t       badPort = 66666;
+    Server::Server badServer(badPort);
+
+    SECTION("Check starting server with bad arguments") { REQUIRE_NOTHROW(badServer.start()); }
+}
+
+TEST_CASE("Testing methods of server services messages")
```

Comment:
You tested only _ping_. Why in description talking amout _methods_?

Reply:
Because in this case we have one more function